core/src/main/scala/org/http4s/HttpDate.scala

```diff
@@ -121,4 +123,164 @@ object HttpDate {
   /** Like `fromZonedDateTime`, but throws any parse failures */
   def unsafeFromZonedDateTime(dateTime: ZonedDateTime): HttpDate =
     unsafeFromInstant(dateTime.toInstant)
+
+  /** `HTTP-date = IMF-fixdate / obs-date` */
+  private[http4s] val parser: Parser1[HttpDate] = {
+    import Parser.{char, failWith, pure, string1}
+    import Rfc5234.{digit, sp}
+
+    def mkHttpDate(
+        year: Int,
+        month: Int,
+        day: Int,
+        hour: Int,
+        min: Int,
+        sec: Int): Parser[HttpDate] =
+      try {
+        val dt = ZonedDateTime.of(year, month, day, hour, min, sec, 0, ZoneOffset.UTC)
+        pure(org.http4s.HttpDate.unsafeFromZonedDateTime(dt))
+      } catch {
+        case _: DateTimeException =>
+          failWith(s"Invalid IMF-fixdate: $year-$month-$day $hour:$min:$sec")
```

Comment:
This demonstrates a failure of our model: the RFC doesn't have anything to say about the date making sense, and being forced to flatMap is bad for parser optimization.

In 1.0, we should carry around the six fields, and have a `Try[ZonedDateTime]`.

Reply:
Ah, nuts:

>    A date-time specification MUST be semantically valid.  That is, the
   day-of-week (if included) MUST be the day implied by the date, the
   numeric day-of-month MUST be between 1 and the number of days allowed
   for the specified month (in the specified year), the time-of-day MUST
   be in the range 00:00:00 through 23:59:60 (the number of seconds
   allowing for a leap second; see [RFC1305]), and the last two digits
   of the zone MUST be within the range 00 through 59.